edge/pkg/edgehub/task/taskexecutor/node_backup.go

```diff
@@ -0,0 +1,86 @@
+package taskexecutor
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/kubeedge/common/constants"
+	commontypes "github.com/kubeedge/kubeedge/common/types"
+	"github.com/kubeedge/kubeedge/edge/cmd/edgecore/app/options"
+	"github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/util"
+	api "github.com/kubeedge/kubeedge/pkg/apis/fsm/v1alpha1"
+	"github.com/kubeedge/kubeedge/pkg/util/fsm"
+	"github.com/kubeedge/kubeedge/pkg/version"
+)
+
+func backupNode(taskReq commontypes.NodeTaskRequest) (event fsm.Event) {
+	event = fsm.Event{
+		Type:   "Backup",
+		Action: api.ActionSuccess,
+	}
+	var err error
+	defer func() {
+		if err != nil {
+			event.Action = api.ActionFailure
+			event.ErrorMsg = err.Error()
+		}
+	}()
+	err = backup()
```

Comment:
Do you need to clear the backup folder after the backup fails?

Reply:
No, because the upgrade is successful does not mean that there is no need to roll back.